### ISSUE TITLE
Zed preventing sleep on MacOS

### ISSUE THREAD

**Issue by sshestopalov** [zed-industries/zed#4580]
### Check for existing issues

- [x] Completed

### Describe the bug / provide steps to reproduce it

Zed editor, when launched, prevents sleep on MacOS. This is the behavior that I personally consider not expected. And I've found my laptop completely discharged in the morning a few times already. 

### Environment

Zed 0.96.3
MacOS 13.5 
Apple M1 Pro

### If applicable, add mockups / screenshots to help explain present your vision of the feature

<img width="1048" alt="image" src="https://github.com/zed-industries/community/assets/26934004/39677cbf-8c2e-4e9d-9265-d5ee659cc0c3">

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by notpeter** [comment]
@sshestopalov Have you observed this recently?

Going through some old issues and I think this may have been fixed somewhere along the way. Thanks

**Comment by sshestopalov** [comment]
@notpeter currently I'm running the 0.142.5 version, and I have no issues with sleep, so I think it was fixed 
<img width="773" alt="image" src="https://github.com/zed-industries/zed/assets/26934004/b2f06444-1ca4-4067-a733-13abaf62ca72">

**Comment by nuno-am** [comment]
Apologies in advance for replying on a closed issue, but for a week or so, I've been having this exact same issue, but it never happened before that.
It could be a coincidence, but around that time I've updated Sonoma to 14.6.1 (MB Pro 2019, 16-inch). I'm currently on Zed 0.149.5 and it's keeping my Mac from sleeping. If Zed is closed, it behaves normally.

**Comment by golovinengineer** [comment]
> Apologies in advance for replying on a closed issue, but for a week or so, I've been having this exact same issue, but it never happened before that.
> 
> It could be a coincidence, but around that time I've updated Sonoma to 14.6.1 (MB Pro 2019, 16-inch). I'm currently on Zed 0.149.5 and it's keeping my Mac from sleeping. If Zed is closed, it behaves normally.
> 
> 
> 
> EDIT: I should've mentioned that the 'Preventing Sleep' column says 'No' for me, but the machine still refuses to sleep and stays working even after I close the lid or choose 'Sleep' on the Apple menu.

I had the same on Monterey, but not sure, it's Zed itself or language server.

**Comment by nuno-am** [comment]
> I had the same on Monterey, but not sure, it's Zed itself or language server.

Yesterday, I decided to switch to VS Code and do the exact same work I was doing with Zed (Vue app development, which requires Node running on the background). With it, there's no additional heat, and my machine sleeps and wakes up normally, even if I leave the Node server running.
I really like Zed, but this is keeping my from using it, as it's also ruining the battery optimization process (it's always stuck at 100% instead of 80% since I noticed this odd behavior).
